src/solve_function.jl

```diff
@@ -0,0 +1,19 @@
+#
+function DEFAULT_LINEAR_SOLVE(A,b,u,p,newA,Pl,Pr,solverdata;kwargs...)
+    solve(LinearProblem(A, b; u0=u); p=p, kwargs...).u
+end
+
+Base.@kwdef struct LinearSolveFunction{F} <: AbstractSolveFunction
+    solve_func::F = DEFAULT_LINEAR_SOLVE
```

Comment:
I don't think there should be a default here. If they use LinearSolveFunction, they are explicitly opting out of using the default, so it would be unsafe to just make it fallback. Instead, it should error and say hey, didn't you wish to provide a function here?